common/test/acceptance/tests/lms/test_lms_edxnotes.py

```diff
@@ -536,6 +535,57 @@ def test_no_content(self):
             "You have not made any notes in this course yet. Other students in this course are using notes to:",
             notes_page_empty.no_content_text)
 
+    def test_notes_works_correctly_with_xss(self):
+        """
+        Scenario: Note text & tags should be HTML and JS escaped
+        Given I am enrolled in a course with notes enabled
+        When I visit the Notes page, with a Notes text & tag containing JS code
+        Then I should see them escaped.
```

Comment:
Just for record keeping i'm posting it here too.
@robrap Thanks for deeper look. 
We are asserting against actual tags because the `.text` property gives us rendered content but the values were properly escaped when  we see them in browser's page source. So in this case 2nd option is true with exception of proper escape.

About your questions:
1 - If i remove escape code from html / underscore the test does fails.

These answers addressed the above concerns and both parties were happy :) ;)